enzy_htp/_interface/pymol_interface.py

```diff
@@ -8,77 +8,193 @@
 Author: Qianzhen (QZ) Shao <shaoqz@icloud.com>
 Date: 2022-02-15
 """
+from pathlib import Path
+from typing import List, Dict, Any, Union, Tuple
 
-from typing import Tuple, List
 
+import pymol2
+import pandas as pd
+
+#TODO(CJ): add something to remove "PyMOL not running. Entering library mode (experimental" message on pymol running
 from enzy_htp import config as eh_config
 from enzy_htp.core import env_manager as em
-from enzy_htp.core import _LOGGER
 from enzy_htp.core import file_system as fs
+from enzy_htp.core import _LOGGER, check_var_type
 from enzy_htp._config.pymol_config import PyMolConfig, default_pymol_config
 from enzy_htp.structure import Structure, PDBParser
 
-# QZ:Env check for package type interface?
-try:
-    import pymol2
-except ImportError:
-    _LOGGER.error(
-        "pymol package is missing. Install with `conda install -c conda-forge open-source-pymol`"
-    )
-
+from .base_interface import BaseInterface
 
-class PyMolInterface:
+class PyMolInterface(BaseInterface):
     """Class that provides a direct inteface for enzy_htp to utilize PyMol package. 
+    
     Atributes:
-        config_ : the PyMolConfig() class which provides settings of PyMol 
+        config_ : The PyMOLConfig() class which provides settings for both running PyMOL and maintaining a compatible environment.
+        env_manager_ : The EnvironmentManager() class which ensures all required environment elements exist.
+        compatible_env_ : A bool() indicating if the current environment is compatible with the object itself.
     """
-
-    def __init__(self, config: PyMolConfig = None) -> None:
-        """Simplistic constructor that optionally takes an PyMolConfig object as its only
-        argument.
+    def __init__(self, parent, config: PyMolConfig = None) -> None:
+        """Simplistic constructor that optionally takes an PyMOLConfig object as its only argument.
+        Calls parent class.
         """
-        self.config_ = config
-        if not self.config_:
-            self.config_ = default_pymol_config()
+        super().__init__(parent, config, default_pymol_config)
+        self.session_ = self.new_session() 
```

Comment:
I dont like the idea that PyMolInterface() object carries a session. Because the PyMolInterface object is singleton if it carries a session all the time the session will be super messed up. This does not fits the design of pymol2. A new sessions should be created everytime for a new independent task. (It is tested to be faster than using a session all the time in one of my previous code running for Lucy)
The session should be supplied as an argument **explicitly** to methods in the interface all the time. (trying to avoid the implicit IO)

Reply:
Removed it.